lib/auth/googleauth.js

```diff
@@ -294,12 +309,25 @@ GoogleAuth.prototype.fromJSON = function(json, opt_callback) {
  */
 GoogleAuth.prototype.fromStream = function(stream, opt_callback) {
   var that = this;
-  var jwtClient = new that.JWTClient();
-  jwtClient.fromStream(stream, function(err) {
-    if (err) {
+  if (!stream) {
+    process.nextTick(function() {
+      callback(
+        opt_callback,
+        new Error('Must pass in a stream containing the service account auth settings.'));
```

Comment:
I may be misreading, but this function looks like it should support both user and service account files but the error message implies that it only supports service accounts.